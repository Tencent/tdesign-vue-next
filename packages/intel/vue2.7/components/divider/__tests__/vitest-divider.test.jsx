/**
 * 该文件由脚本自动生成，如需修改请联系 PMC
 * This file generated by scripts of tdesign-api. `npm run api:docs Divider Vue(PC) vitest,finalProject`
 * If you need to modify this file, contact PMC first please.
 */
import { mount } from '@vue/test-utils';
import { Divider } from '..';

describe('Divider Component', () => {
  ['left', 'right', 'center'].forEach((item) => {
    it(`props.align is equal to ${item}`, () => {
      const wrapper = mount({
        render() {
          return <Divider align={item}>Text</Divider>;
        },
      });
      expect(wrapper.classes(`t-divider--with-text-${item}`)).toBeTruthy();
      expect(wrapper.element).toMatchSnapshot();
    });
  });

  it('props.content works fine', () => {
    const wrapper = mount({
      render() {
        return <Divider content={(h) => <span class="custom-node">TNode</span>}></Divider>;
      },
    });
    expect(wrapper.find('.custom-node').exists()).toBeTruthy();
    expect(wrapper.element).toMatchSnapshot();
  });

  it('slots.content works fine', () => {
    const wrapper = mount({
      render() {
        return <Divider scopedSlots={{ content: () => <span class="custom-node">TNode</span> }}></Divider>;
      },
    });
    expect(wrapper.find('.custom-node').exists()).toBeTruthy();
    expect(wrapper.element).toMatchSnapshot();
  });

  it('props.dashed works fine', () => {
    // dashed default value is false
    const wrapper1 = mount({
      render() {
        return <Divider></Divider>;
      },
    });
    expect(wrapper1.classes('t-divider--dashed')).toBeFalsy();
    // dashed = true
    const wrapper2 = mount({
      render() {
        return <Divider dashed={true}></Divider>;
      },
    });
    expect(wrapper2.classes('t-divider--dashed')).toBeTruthy();
    // dashed = false
    const wrapper3 = mount({
      render() {
        return <Divider dashed={false}></Divider>;
      },
    });
    expect(wrapper3.classes('t-divider--dashed')).toBeFalsy();
  });

  it('props.default works fine', () => {
    const wrapper = mount({
      render() {
        return <Divider default={(h) => <span class="custom-node">TNode</span>}></Divider>;
      },
    });
    expect(wrapper.find('.custom-node').exists()).toBeTruthy();
    expect(wrapper.element).toMatchSnapshot();
  });

  it('slots.default works fine', () => {
    const wrapper = mount({
      render() {
        return <Divider scopedSlots={{ default: () => <span class="custom-node">TNode</span> }}></Divider>;
      },
    });
    expect(wrapper.find('.custom-node').exists()).toBeTruthy();
    expect(wrapper.element).toMatchSnapshot();
  });

  ['horizontal', 'vertical'].forEach((item) => {
    it(`props.layout is equal to ${item}`, () => {
      const wrapper = mount({
        render() {
          return <Divider layout={item}></Divider>;
        },
      });
      expect(wrapper.classes(`t-divider--${item}`)).toBeTruthy();
      expect(wrapper.element).toMatchSnapshot();
    });
  });
});
