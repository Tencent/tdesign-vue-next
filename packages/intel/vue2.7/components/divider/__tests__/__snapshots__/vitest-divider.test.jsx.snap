// Vitest Snapshot v1

exports[`Divider Component > props.align is equal to center 1`] = `
<div
  class="t-divider t-divider--horizontal t-divider--with-text t-divider--with-text-center"
>
  <span
    class="t-divider__inner-text"
  >
    Text
  </span>
</div>
`;

exports[`Divider Component > props.align is equal to left 1`] = `
<div
  class="t-divider t-divider--horizontal t-divider--with-text t-divider--with-text-left"
>
  <span
    class="t-divider__inner-text"
  >
    Text
  </span>
</div>
`;

exports[`Divider Component > props.align is equal to right 1`] = `
<div
  class="t-divider t-divider--horizontal t-divider--with-text t-divider--with-text-right"
>
  <span
    class="t-divider__inner-text"
  >
    Text
  </span>
</div>
`;

exports[`Divider Component > props.content works fine 1`] = `
<div
  class="t-divider t-divider--horizontal t-divider--with-text t-divider--with-text-center"
>
  <span
    class="t-divider__inner-text"
  >
    <span
      class="custom-node"
    >
      TNode
    </span>
  </span>
</div>
`;

exports[`Divider Component > props.default works fine 1`] = `
<div
  class="t-divider t-divider--horizontal t-divider--with-text t-divider--with-text-center"
>
  <span
    class="t-divider__inner-text"
  >
    <span
      class="custom-node"
    >
      TNode
    </span>
  </span>
</div>
`;

exports[`Divider Component > props.layout is equal to horizontal 1`] = `
<div
  class="t-divider t-divider--horizontal"
/>
`;

exports[`Divider Component > props.layout is equal to vertical 1`] = `
<div
  class="t-divider t-divider--vertical"
/>
`;

exports[`Divider Component > slots.content works fine 1`] = `
<div
  class="t-divider t-divider--horizontal t-divider--with-text t-divider--with-text-center"
>
  <span
    class="t-divider__inner-text"
  >
    <span
      class="custom-node"
    >
      TNode
    </span>
  </span>
</div>
`;

exports[`Divider Component > slots.default works fine 1`] = `
<div
  class="t-divider t-divider--horizontal t-divider--with-text t-divider--with-text-center"
>
  <span
    class="t-divider__inner-text"
  >
    <span
      class="custom-node"
    >
      TNode
    </span>
  </span>
</div>
`;
